Packages/com.unity.inputsystem/InputSystem/Plugins/Users/InputUser.cs

```diff
@@ -1506,10 +1506,17 @@ private static void OnDeviceChange(InputDevice device, InputDeviceChange change)
                 // New device was added. See if it was a device we previously lost on a user.
                 case InputDeviceChange.Added:
                 {
-                    // Could be a previously lost device. Could affect multiple users. Repeatedly search in
-                    // s_AllLostDevices until we can't find the device anymore.
-                    var deviceIndex = s_GlobalState.allLostDevices.IndexOfReference(device, s_GlobalState.allLostDeviceCount);
-                    while (deviceIndex != -1)
+                    int FindLostDevice(int deviceIndex, int deviceId)
+                    {   // Note that we compare by Device ID since the lost device may be another instance
+                        // since its recreated during layout override. For other cases its the same reference.
+                        return s_GlobalState.allLostDevices.IndexOf((d) => d.deviceId == device.deviceId,
+                            deviceIndex, s_GlobalState.allLostDeviceCount);
+                    }
```

Comment:
This local method will likely lead to unnecessary garbage on the GC heap (as it references the `device` local variable).